### DISCUSSION TITLE
Not sharing an IPP printer

### DISCUSSION THREAD

**Discussion by debiantriage** [OpenPrinting/gutenprint-printer-app#2] [Ideas]
Of its very nature a network-capable IPP printer is published over the network when exposed on an ethernet or wireless interface. However, it is observed that most network-capable IPP printer also provide the IPP-over-USSB protocol, allowing a user to restrict publishing to the localhost interface. In other word, a user has the choice of sharing a printer or not.

A  Printer Application advertises itself on all interfaces and there does not appear to be a way within the Application itself to restrict this. A user who does not want a device to be available on ethernet or wireless interfaces might consider the following to provide a solution:

- Edit **/etc/avahi/avahi-daemon.conf** to have `deny-interfaces= INTERFACE_NAME0`. INTERFACE_NAME0 may be obtained from `ip a`.
- `systemctl restart avahi-daemon.service avahi-daemon.socket`.
-  `systemctl restart snap.gutenprint-printer-app.gutenprint-printer-app-server.service`.

The outcome:
```
brian@test-new:~$ avahi-browse -t _ipp._tcp
+     lo IPv4 420                                           Internet Printer     local
```

**Comment by tillkamppeter** [comment]
Thanks for the hint. You should bring it up on the [PAPPL repo](https://github.com/michaelrsweet/pappl/) repo, and there as an issue (bug? feature?), to ask for an easy way to switch it to localhost-only and/or to also switch on and off (un)wished network interfaces.

**Reply by debiantriage** [reply]
[Mentioned](https://github.com/michaelrsweet/pappl/discussions/278#discussioncomment-6356223) at PAPPL Discussions.
